scripts/generate_category_bundles.py

```diff
@@ -0,0 +1,422 @@
+        lines.append("")  # spacer
+        lines.append(p.body.strip())
+        lines.append("")  # spacer
+    with open(out_path, "w", encoding="utf-8") as f:
+        f.write("\n".join(lines))
+
+
+# ----------------------------
+# Main
+# ----------------------------
+
+def build_category_bundles(config_path: str, fmt: str, dry_run: bool, limit: int,
+                           token_estimator: str):
+    config = load_config(config_path)
+    repo_root = Path(__file__).resolve().parent.parent
+    ai_dir = resolve_ai_dir(repo_root, config)
+    raw_base = build_raw_base(config)
+
+    categories_order, base_cats = categories_from_config(config)
+    if not categories_order:
+        print("[!] No categories_order configured under content.categories_order.")
+        return
+
+    pages = load_all_pages(ai_dir)
+
+    # Optional: limit pages during dry-runs
+    if limit > 0:
+        pages = pages[:limit]
+
+    # Precompute token counts once per page
+    page_tokens: Dict[str, int] = {p.slug: estimate_tokens(p.body, token_estimator) for p in pages}
+
+    out_root = (repo_root / config.get("outputs", {}).get("public_root", "/.ai/").strip("/") / "categories").resolve()
+
+    if dry_run:
+        print(f"[dry-run] ai_dir={ai_dir}")
+        print(f"[dry-run] total_pages={len(pages)}")
+        print(f"[dry-run] categories={categories_order}")
+        print(f"[dry-run] base_context={base_cats}")
+        print(f"[dry-run] token_estimator={token_estimator}")
+        if pages:
+            sample = pages[0]
+            print(f"[dry-run] sample_page: slug={sample.slug} title={sample.title} cats={sample.categories} est_tokens={page_tokens.get(sample.slug)}")
+        print(f"[dry-run] output_dir={out_root}")
+        print(f"[dry-run] raw_base={raw_base}")
+
+    # Precompute base context page sets
+    base_sets: List[List[AiPage]] = [select_pages_for_category(c, pages) for c in base_cats]
+    base_union: List[AiPage] = union_pages(base_sets)
+
+    # Generate bundles
+    for cat in categories_order:
+        cat_slug = slugify_category(cat)
+        is_base = cat in base_cats
+
+        if is_base:
+            # Standalone base bundle: just this category's pages
+            pages_cat = select_pages_for_category(cat, pages)
+            pages_out = sorted(pages_cat, key=lambda p: p.title.lower())
+
+            if dry_run:
+                print(f"[dry-run] base bundle: {cat} ({len(pages_out)} pages)")
+            else:
+                out_root.mkdir(parents=True, exist_ok=True)
+                if fmt in ("manifest", "all"):
+                    write_manifest(out_root / f"{cat_slug}.manifest.json", cat, cat_slug, False, base_cats, pages_out, raw_base, token_estimator, page_tokens)
+                if fmt in ("jsonl", "all"):
+                    write_jsonl(out_root / f"{cat_slug}.bundle.jsonl", pages_out, raw_base, token_estimator, page_tokens)
+                if fmt in ("md", "all"):
```

Comment:
Ah ok, so we can generate these file types. So I guess this is my biggest concern and what I think we should document in the /.ai/README.md.

Why do we need all these different files? What do each of them offer? I thought I saw these at one point and now they're not there anymore, which is why I previously commented about removing stuff related to these. So obviously I'm confused hahaha what's the plan here?